pyproject.toml

```diff
@@ -17,31 +17,30 @@ readme = "README.md"
 dynamic = ["version"]
 
 dependencies = [
-    'numpy',
-    'scipy',
-    'matplotlib',
-    'PyQt5',
-    'pandas',
-    'pyserial',
+    "numpy >=2, <3",
+    "scipy >=1, <2",
+    "matplotlib >=3, <4",
+    "PyQt5 >=5, <6",
+    "pandas >=2, <3",
+    "pyserial >=3, <4",
     "pyOSC3@git+https://github.com/glopesdev/pyosc3.git@master",
     "newscale@git+https://github.com/AllenNeuralDynamics/python-newscale@axes-on-target",
     "aind-auto-train@git+https://github.com/AllenNeuralDynamics/aind-foraging-behavior-bonsai-automatic-training.git@main",
     "aind-slims-api@git+https://github.com/AllenNeuralDynamics/aind-slims-api@main",
     "aind-dynamic-foraging-models@git+https://github.com/AllenNeuralDynamics/aind-dynamic-foraging-models@main",
-    "pynwb",
-    "requests",
-    "harp-python",
-    "openpyxl",
-    "deepdiff",
+    "pynwb >=2, <3",
+    "requests >=2, <3",
+    "harp-python >=0.3, <2",
+    "openpyxl >=3, <4",
+    "deepdiff >=8, <9",
     "aind-data-schema==1.1.0",
     "aind-data-schema-models==0.5.6",
-    "pydantic==2.9.2",
```

Comment:
Can you keep the versions that were already set for pydantic, python-logging-loki, and pykeepass?

Reply:
Why do you want to pin at a particular version? Pedantic likely follows semantic versioning, presumably any version 2.9.2 or higher in the same compatible range (i.e. `[2.9.2, 3)`) should work.

Reply:
Basically anything in that range should still have the features you want (presumably something happened in pydantic exactly at 2.9.2 that this project depends on) while still maintaining a backwards-compatible API, and you will get bug fixes etc. for free. Unless they are not adhering to semantic versioning, and you know that 2.9.3 breaks your code (!) then I think it's bad practice to pin a specific version. It just makes the project unnecessarily difficult to install. I understand that many aind packages (e.g. data schema) do not adhere to semantic versioning and break backwards compatibility in compatible version ranges, so I have left those pinned at a single version.

Reply:
The versions of python-logging-loki and pykeepass were added by Ross recently, and I assume he knows what he is doing. (I guess I dont really understand the difference between ~= and <, > pairs)

you are right that we just need greater than 2.9.2 for pydantic.

Reply:
`~=2.9.2` is equivalent to `>=2.9.2, <3`, but since I was not able to use `~=` locally I changed everything to use the more verbose but equivalent format. Semantically, I did not change the compat range for python-logging-loki or pykeepass, I only changed the syntax.